cmd/infra/aws/create.go

```diff
@@ -124,102 +145,56 @@ func (o *CreateInfraOptions) CreateInfra() (*CreateInfraOutput, error) {
 		Name:        o.Name,
 		BaseDomain:  o.BaseDomain,
 	}
-	client, err := ec2Client(o.AWSCredentialsFile, o.Region)
-	if err != nil {
-		return nil, err
-	}
-	result.Zone, err = o.firstZone(client)
+	result.Zone, err = o.firstZone(o.EC2Client)
 	if err != nil {
 		return nil, err
 	}
-	result.VPCID, err = o.createVPC(client)
+	result.VPCID, err = o.createVPC(o.EC2Client)
 	if err != nil {
 		return nil, err
 	}
-	if err = o.CreateDHCPOptions(client, result.VPCID); err != nil {
-		return nil, err
-	}
-	result.PrivateSubnetID, err = o.CreatePrivateSubnet(client, result.VPCID, result.Zone)
-	if err != nil {
+	if err = o.CreateDHCPOptions(o.EC2Client, result.VPCID); err != nil {
 		return nil, err
 	}
-	result.PublicSubnetID, err = o.CreatePublicSubnet(client, result.VPCID, result.Zone)
+	result.PrivateSubnetID, err = o.CreatePrivateSubnet(o.EC2Client, result.VPCID, result.Zone)
```

Comment:
do we need to pass the client if it's now part of the struct?

Reply:
No, but I didn't feel like changing the signatures and I also generally don't really like relying on too much state so wouldn't be opposed to keeping the method closer to stateless anyway